pkg/apis/awsnodetemplate/v1alpha1/awsnodetemplate_validation.go

```diff
@@ -28,5 +33,18 @@ func (a *AWSNodeTemplate) Validate(ctx context.Context) (errs *apis.FieldError)
 }
 
 func (a *AWSNodeTemplateSpec) validate(ctx context.Context) (errs *apis.FieldError) {
-	return a.AWS.Validate()
+	return errs.Also(
+		a.AWS.Validate(),
+		a.validateUserData(),
+	)
+}
+
+func (a *AWSNodeTemplateSpec) validateUserData() (errs *apis.FieldError) {
+	if a.UserData == nil {
+		return nil
+	}
+	if a.LaunchTemplateName != nil {
```

Comment:
It strikes me that we should simply disallow LaunchTemplateName in AWSNodeTemplate.

Reply:
I don't think we can at this point. It's already released and that would break backwards compatibility.  I realize we could in the confines of semver, but I don't think it's worth it right now and we should just wait for v1alpha6 provisioner and increment the AWSNodeTemplate version at that point.

Reply:
Discussed this offline, and it's worth considering this during a single version bump (in alignment with whenever we do `v1alpha6`) but let's leave it in until then.